src/library/commissioner_impl.cpp

```diff
@@ -445,24 +445,41 @@ void CommissionerImpl::SetCommissionerDataset(ErrorHandler aHandler, const Commi
 }
 
 void CommissionerImpl::GetActiveDataset(Handler<ActiveOperationalDataset> aHandler, uint16_t aDatasetFlags)
+{
+    auto rawDatasetHandler = [aHandler](const ByteArray *aRawDataset, Error aError) {
+        Error                    error;
+        ActiveOperationalDataset dataset;
+
+        SuccessOrExit(error = aError);
+        SuccessOrExit(error = DecodeActiveOperationalDataset(dataset, *aRawDataset));
+        VerifyOrExit(dataset.mPresentFlags & ActiveOperationalDataset::kActiveTimestampBit,
+                     error = ERROR_BAD_FORMAT("Active Timestamp is not included in MGMT_ACTIVE_GET.rsp"));
+
+        aHandler(&dataset, error);
+    exit:
+        if (error != ErrorCode::kNone)
+        {
+            aHandler(nullptr, error);
+        }
+    };
+
+    GetRawActiveDataset(rawDatasetHandler, aDatasetFlags);
+}
+
+void CommissionerImpl::GetRawActiveDataset(Handler<ByteArray> aHandler, uint16_t aDatasetFlags)
 {
     Error         error;
     coap::Request request{coap::Type::kConfirmable, coap::Code::kPost};
     ByteArray     datasetList = GetActiveOperationalDatasetTlvs(aDatasetFlags);
 
     auto onResponse = [aHandler](const coap::Response *aResponse, Error aError) {
-        Error                    error;
-        ActiveOperationalDataset dataset;
+        Error error;
 
         SuccessOrExit(error = aError);
         VerifyOrExit(aResponse->GetCode() == coap::Code::kChanged,
                      error = ERROR_BAD_FORMAT("expect CoAP::CHANGED for MGMT_ACTIVE_GET.rsp message"));
 
-        SuccessOrExit(error = DecodeActiveOperationalDataset(dataset, *aResponse));
-        VerifyOrExit(dataset.mPresentFlags & ActiveOperationalDataset::kActiveTimestampBit,
-                     error = ERROR_BAD_FORMAT("Active Timestamp is not included in MGMT_ACTIVE_GET.rsp"));
-
-        aHandler(&dataset, error);
+        aHandler(&aResponse->GetPayload(), error);
```

Comment:
According to Thread Spec., a message can contains extra TLVs and implementations are supposed to ignore these TLVs if unable to handle. So, I would suggest to only get the bytes of the Active Dataset TLV if it's easy to do.

Reply:
Yes, extra TLVs are ignored when decoding the binary blob to struct `ActiveOperationalDataset`, but I think we should return what we get here, whether the receiver of the binary blob is going to ignore or reinterpret the extra TLVs is not limited by the commissioner.